### PR TITLE
Fix parsing of At_Root_Blocks with expressions

### PR DESCRIPTION
**Pull request by mgreter**
Fixes https://github.com/sass/libsass/issues/1651
Spec: https://github.com/sass/sass-spec/pull/575

Does not pass the error spec. Couldn't come up with anything better, but it should solve the reported regression. We could need some more spec tests for this ...

### CODE REVIEW

**Comment by xzyfer** [comment]
Ok so this is more complicated than it seems.

We do *fully* support `without` and there is good test coverage for it. The at-root logic happens in the cssize step. This is how Ruby Sass does it.

We should not throw the `@extend` error in the parser. We should throw during the extend process. Only then can we know for sure that the `@extend` is valid.

File: src/parser.cpp [hunk 1]
```diff
@@ -2121,7 +2121,11 @@ namespace Sass {
       expr = parse_at_root_expression();
     }
     if (peek < exactly<'{'> >()) {
-      body = parse_block(true);
+      std::string feat("");
+      if (expr && expr->feature())
+        feat = expr->feature()->to_string();
+      // allow declarations in certain cases
+      body = parse_block(feat != "without");
```
Comment (xzyfer) [inline src/parser.cpp]: Unfortunately the `without` directive is too complex for this. For example should error.

```
a { b: c }

.d {
  @supports (foo: bar) {
    @media all {
      @at-root (without: media) {
        @extend a;
      }
    }
  }
}
```

**Comment by mgol** [comment]
This still has the `3.3.3` milestone and libsass `3.3.4` seems to already have been released. Why is that?

**Comment by xzyfer** [comment]
This patch slated for 3.3.3 but it isn't sufficient. Closing as this approach doesn't address the underlying issue.

**Pull request closed without merging**